### PR TITLE
[Needs docs] Add optimizer to config

### LINKED ISSUE #267: Learning rate is in a line of hard code
The learning rate in the litpose is set to 1e-3 in `lightning_pose/models/base.py`. It would be great to make [learning rate](https://github.com/paninski-lab/lightning-pose/blob/main/lightning_pose/models/base.py#L408C1-L425) can be specified in the config training file. 

Moreover, it would be nice to select different optimizer/scheduler. Thanks!

### PR DESCRIPTION
**Pull request by ksikka**
Resolves #267

### CODE REVIEW

**Comment by ksikka** [comment]
@themattinthehatt ready for initial review. if the interface looks good to you, I'll make docs changes.

File: lightning_pose/models/heatmap_tracker.py [hunk 1]
```diff
@@ -66,6 +66,8 @@ def __init__(
         pretrained: bool = True,
         output_shape: tuple | None = None,  # change
         torch_seed: int = 123,
+        optimizer: str = "Adam",
+        optimizer_params: DictConfig | dict | None = None,
```
Comment (themattinthehatt) [inline lightning_pose/models/heatmap_tracker.py:70]: add docstrings

**Review by themattinthehatt — APPROVED** [review]

File: lightning_pose/models/heatmap_tracker.py (hunk 1 above)
Comment (ksikka) [reply lightning_pose/models/heatmap_tracker.py:70]: Done
